timeboost-sequencer/src/queue.rs

```diff
@@ -57,50 +57,50 @@ impl TransactionQueue {
         self.0.lock().index = idx
     }
 
-    pub fn add_transactions<I>(&self, it: I)
+    pub fn add_bundles<I>(&self, it: I)
     where
-        I: IntoIterator<Item = Transaction>,
+        I: IntoIterator<Item = BundleVariant>,
     {
         let time = Timestamp::now();
         let now = Instant::now();
 
         let mut inner = self.0.lock();
 
         inner.set_time(time);
-
-        for mut t in it.into_iter() {
-            if let Ok(kid) = KeysetId::try_from(t.data().as_ref()) {
-                t.set_keyset(kid);
-            } else {
-                continue;
-            }
-
-            if t.to() != &inner.priority_addr {
-                inner.transactions.push_back((now, t));
-                continue;
-            }
-
-            // TODO: Check transaction signature is valid PLC signature.
-
-            let epoch = t.nonce().to_epoch();
-
-            if epoch < inner.time.epoch() || epoch > inner.time.epoch() + 1 {
-                continue;
+        let epoch_now = inner.time.epoch();
+        for b in it.into_iter() {
+            match b {
+                BundleVariant::Regular(b) => inner.regular.push_back((now, b)),
+                BundleVariant::Priority(b) =>
+                // TODO: Check auction contract address on bundle
+                {
+                    match b.validate(epoch_now, Some(inner.priority_addr)) {
+                        Ok(_) => {
+                            let epoch = b.bundle().epoch();
+                            inner.priority.entry(epoch).or_default().push(b);
+                        }
+                        Err(e) => {
+                            trace!(
+                                "Validation failed for bundle signed by: {:?}, error: {:?}",
```

Comment:
No need to do string interpolation. `trace!` accepts structured parameters, e.g.

```rust
trace!(signer = ?b.sender(), err = %e, "bundle validation failed")
```